.github/workflows/per-pr.yml

```diff
@@ -37,6 +37,17 @@ jobs:
       - name: Run flag test
         run: ${{ github.workspace }}/scripts/test-proxy-startup-with-env-vars.sh
 
+  # This ensures that std output generated when running binary with `--help` is reflected in
+  # in docs/help_out.txt
+  help-output-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: ${{ env.GO_VERSION }}
+      - run: make gen-static-help-output && git diff --exit-code
```

Comment:
So we'll have to keep this not mandatory to be able to merge PRs that are adding new flags or modifying flag messages.

But will CI show a red X? Or does it not for non-mandatory jobs?

Reply:
> So we'll have to keep this not mandatory to be able to merge PRs that are adding new flags or modifying flag messages.

nope - you'll just need to run `make gen-static-help-output` to refresh the file so its up-to-date. This check ensures that the `docs/help_output.txt` is always reflective of the latest binary which will ensure any diffs to flags are captured in PRs going forward

Reply:
it will still show a red X but will be mergable iiuc